CMakeLists.txt

```diff
@@ -106,6 +106,11 @@ if(NOT WIN32)
 endif()
 string(REPLACE ";" " " CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
 
+# Use RUNPATH instead of RPATH for all shared libs and executables on Linux
+if(IS_LINUX)
+  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--enable-new-dtags")
+  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--enable-new-dtags")
+endif()
```

Comment:
Prefer to use RUNPATH over RPATH. If RPATH is used, then we cannot override it during execution since LD_LIBRARY_PATH has lower precedence. The order of precedence for search paths on Linux are:

1- RPATH
2- LD_LIBRARY_PATH
3- RUNPATH